src/warc_writer.rs

```diff
@@ -24,7 +24,14 @@ impl<W: Write> WarcWriter<W> {
     /// Write a single record.
     ///
     /// The number of bytes written is returned upon success.
-    pub fn write(&mut self, record: &Record) -> io::Result<usize> {
+    pub fn write(&mut self, record: Record) -> io::Result<usize> {
```

Comment:
Do we want `write` to consume the `record` value?

Reply:
I am on the fence about that.

I didn't change it yet, because it would make several of the code examples more annoying to write, before I continued my work beyond this point. That will not be a concern for my next update, though.

The main advantage of consuming the record is that it would be much more efficient based on what I have now.

The main argument against it is that it is an ergonomic barrier. You are forced to iterate over collections destructively, for example:

```
let records: Vec<Record> = generate_my_records();
records.iter().for_each(|record| writer.write(record)); // ERROR: expect Record, found &Record
```

You also couldn't write them multiple times without cloning:
```
let record: Record = ...;
stream1.write(record);
stream2.write(record); // ERROR: Record used after move
```

Given that each record currently contains a body -- which even in my use case could be in excess of 8 MB -- requiring the user to make copies in fairly common cases seems bad.

I am torn, because -- spoiler alert -- my plan to fix #13 is to split `body` out of `Record` entirely, which will make doing a `clone()` not nearly so expensive.

I am not sure whether it is worth rework to fix that prior to then or not. Your thoughts?

Reply:
> The main argument against it is that it is an ergonomic barrier. You are forced to iterate over collections destructively, for example:

👍 

> I am not sure whether it is worth rework to fix that prior to then or not. Your thoughts?

For now I think it makes sense that `write` doesn't take ownership of the record value and it's passed by reference.